pages/applications/view/[id]/index.tsx

```diff
@@ -56,8 +56,28 @@ export default function ApplicationPage({
   data,
   history,
 }: PageProps): JSX.Element | null {
-  if (!data.id) return <Custom404 />;
   const router = useRouter();
```

Comment:
I'm pretty sure that `useRouter` is a state based hook, which means that every time that `router.query.tab` changes, this component will be re-rending anyway. If I'm right, then I think you could skip the state all together and just do:

```
  const tab = router.query.tab ?? 'overview';

  const handleTabChange = (newValue: string) => {
    router.push(`/applications/view/${data.id}?tab=${newValue}`, undefined, {
      shallow: true,
    });
  };
```

And each time you call handleTabChange you should be able to rely on tab updating. In other words, router.push is internally doing a setState so you probably don't need a second one yourself.

Reply:
Makes sense. Have updated. We could probably use this shallow routing on other places in the admin but maybe for another PR.